crates/distances/tests/test_sets.rs

```diff
@@ -27,3 +29,92 @@ fn test_dice() {
     let distance: f32 = dice(&x, &y);
     assert!((distance - 0.8) < f32::EPSILON);
 }
+
+/// Generates a length 1000 set with each bit flipped randomly on or off
+fn gen_set() -> Vec<u16> {
+    let mut vec = Vec::new();
+    for i in 0..1000 {
+        if rand::random() {
+            vec.push(i);
+        }
+    }
+    vec
+}
+
+/// Random exhaustive testing of set distances, manually creating union and intersection values
+#[test]
+fn sets_test() {
+    for _ in 0..10000 {
+        let x: Vec<u16> = gen_set();
+        let y: Vec<u16> = gen_set();
+        let mut union: f32 = 0.0;
+        let mut intersection: f32 = 0.0;
+        let mut size: f32 = 0.0;
```

Comment:
Not a big deal, but I think it's usually better to give these variables types that make sense based on what they store, and then cast to a different type when the need arises. For example, `union`, `intersection`, and `size` will always be non-negative integers less than 2000, so a usize type would suffice. 

The usize type will cause you problems when you start doing division, but you can get around that by doing something like f32::from(intersection) at that point, rather than preemptively making those variables f32s.